src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -7,11 +7,21 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
+    internal static class FileTextLoaderOptions
+    {
+        // File Threshold 100MB
+        [ExportOption]
+        internal static readonly Option<long> FileLengthThreshold = new Option<long>(nameof(FileTextLoaderOptions), nameof(FileLengthThreshold), defaultValue: 100 * 1024 * 1024,
+            storageLocations: new LocalUserProfileStorageLocation(@"Roslyn\Internal\Performance\Text\FileLengthThreshold"));
```

Comment:
The use of "Internal" here seemed wrong given users may need to change it.  Can't tell if this is an established pattern though.  If it is then okay.